script.module.clouddrive.common/clouddrive/common/service/source.py

```diff
@@ -91,148 +100,271 @@ def get_size(self, size):
     
     def get_cloud_drive_addons(self):
```

Comment:
What is the need/use of this function? THX

Reply:
I have a source mode in the addons. 
You can add your drive as a source in Kodi by pointing to http://localhost:8585/source and you will be able to simulate a web directory.
The first page in that navigation is the list of addons supporting this feature, which I discover by checking the list of addons dependant on this common module.

Reply:
This will require further review :)

Reply:
No worries. Let me know if more explanation is necessary.